### PR TITLE
#167649013 Landville admin and client admin can update and delete company

### PR DESCRIPTION
**Pull request by bl4ck4ndbr0wn**
## Description ##
- create permission class to allow admin update and delete any client company

## Type of change ##
Please select the relevant option
- [ ] Bug fix(a non-breaking change which fixes an issue)
- [x] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of a feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How to test this PR
- Start the development server and send the following through postman

*Fetch Client Company as Lanville Admin*
```
_URL:(GET) http://localhost:8000/api/v1/auth/client/2/
headers: {
Authorization: {{token}}
}
```
*Update client company as Client Admin*
```
_URL:(PATCH)  http://localhost:8000/api/v1/auth/client/
BODY:
{
  "client_name": "safaricom",
  "phone": "+234 123 4567890",
  "email": "alpha@mailinator.com",
  "address": {
  	"Street": "Roysambu",
  	"City": "Nairobi",
  	"State": "Nairobi"
  }
}

```
*Update client company as Lanville Admin*
```
_URL:(PATCH)  http://localhost:8000/api/v1/auth/client/2/
BODY:
{
  "client_name": "safaricom",
  "phone": "+234 123 4567890",
  "email": "alpha@mailinator.com",
  "address": {
  	"Street": "Roysambu",
  	"City": "Nairobi",
  	"State": "Nairobi"
  }
}

```

*Delete client company as Client Admin*
```
_URL:(DELETE)  http://localhost:8000/api/v1/auth/client/

```

*Delete client company as Lanville Admin*
```
_URL:(DELETE)  http://localhost:8000/api/v1/auth/client/2/

```

## How Has This Been Tested? ##
Please describe the tests that you ran to verify your changes
- [ ] End to end test
- [x] Integration test
- [x] unit test

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## Related Pivotal Tracker stories ##
[#167649013](https://www.pivotaltracker.com/story/show/167649013)

### CODE REVIEW

File: authentication/views.py [hunk 1]
```diff
@@ -106,7 +108,8 @@ def get(self, request):
         try:
             payload = jwt.decode(token, settings.SECRET_KEY)
         except jwt.exceptions.DecodeError:
-            return HttpResponseRedirect(domain+'?verified_status=invalid_link')  # noqa
+            return HttpResponseRedirect(
+                domain + '?verified_status=invalid_link')  # noqa
```
Comment (espoirMur) [inline authentication/views.py:112]: String formating is more pythonic here bro, 

something like  `'{}format this'.format(value)`

File: authentication/views.py [hunk 2]
```diff
@@ -219,11 +222,12 @@ class ClientCreateView(generics.GenericAPIView, BaseUtils):
     Register new client company by client admin
     """
     serializer_class = ClientSerializer
-    renderer_classes = (UserJSONRenderer,)
+    renderer_classes = (ClientJSONRenderer,)
```
Comment (espoirMur) [inline authentication/views.py:225]: Why are you changing the renderer?

**Review by espoirMur — CHANGES_REQUESTED** [review]
Good work bro,

File: authentication/views.py [hunk 3]
```diff
@@ -285,21 +288,125 @@ def post(self, request):
...
+        """
+        data = request.data
+        company = self.get_queryset()
+        serializer = self.serializer_class(company[0], data=data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        response = {
+            "client_company": serializer.data,
+            "message": "Successfully Update to company."
```
Comment (CryceTruly) [inline authentication/views.py]: Great work @bl4ck4ndbr0wn, please check the grammar here.

**Review by CryceTruly — CHANGES_REQUESTED** [review]

File: authentication/views.py (hunk 2 above)
Comment (bl4ck4ndbr0wn) [reply authentication/views.py:225]: This allows easier and dynamic responses to the data being sent to the user.

File: authentication/views.py (hunk 1 above)
Comment (bl4ck4ndbr0wn) [reply authentication/views.py:112]: That's not my feature. That was already done along time ago.

File: authentication/views.py (hunk 3 above)
Comment (bl4ck4ndbr0wn) [reply authentication/views.py]: Done.

**Review by malep2007 — CHANGES_REQUESTED** [review]
@bl4ck4ndbr0wn, Squash the django migration files. You can read up from [here](https://www.techiediaries.com/squash-django-migrations/)

**Review by malep2007 — APPROVED** [review]

**Review by CryceTruly — APPROVED** [review]

**Review by espoirMur — APPROVED** [review]

**Review by patrickf949 — APPROVED** [review]
LGTM

**Review by Victoradukwu — CHANGES_REQUESTED** [review]
@bl4ck4ndbr0wn Please add a section on how to test this PR

**Comment by bl4ck4ndbr0wn** [comment]
> @bl4ck4ndbr0wn Please add a section on how to test this PR

Done

**Comment by malep2007** [comment]
@bl4ck4ndbr0wn , please resolve conflict here

**Comment by bl4ck4ndbr0wn** [comment]
> @bl4ck4ndbr0wn , please resolve conflict here

Done.

**Review by Victoradukwu — APPROVED** [review]
LGTM

**Pull request merged**